sqlccl: correctly mutate table descriptors during resumed RESTORE

madelynnblue opened the pull request:
If a RESTORE was resumed after a pause or node failure, it would
attempt to regenerate the sql descriptors it was using. However it did
this in a way that was different than the first invocation that didn't
use the same resume codepath. This mutation was to remove the FK on
indexes if the user requested it. Move the mutation to a downstream
place so both invocation paths will mutate index descriptors in the
same way.

Also improve some test infra to return errors more quickly and with
useful messages.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20092)
<!-- Reviewable:end -->

madelynnblue committed:
sqlccl: correctly mutate table descriptors during resumed RESTORE

If a RESTORE was resumed after a pause or node failure, it would
attempt to regenerate the sql descriptors it was using. However it did
this in a way that was different than the first invocation that didn't
use the same resume codepath. This mutation was to remove the FK on
indexes if the user requested it. Move the mutation to a downstream
place so both invocation paths will mutate index descriptors in the
same way.

Also improve some test infra to return errors more quickly and with
useful messages.

mjibson commented:
Ping @dt.

benesch commented on pkg/ccl/sqlccl/backup_test.go:
```diff
@@ -619,18 +619,32 @@ func TestBackupRestoreCheckpointing(t *testing.T) {
+	if jobFailedErr != nil {
+		return jobFailedErr
+	}
+	return err
```
Ah, much better!

benesch reviewed (approved):
Nice, LGTM!

dt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> nice catch!

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20092#-:-KzPqpZ9DjMvYjVgZzcK:b-pe67a)*
<!-- Sent from Reviewable.io -->
